<?xml version="1.0"?>
<launch>

  <param name="linear_scale" type="double" value="0.2" />
  <param name="distance_constante" type="double" value="0.3" />

  <include file="$(find challenge_project)/launch/gazebo.launch">
  <arg name="x" value="0"/>
  <arg name="y" value="0"/>
  <arg name="z" value="0.0175"/>
  <arg name="roll" value="0"/>
  <arg name="pitch" value="0"/>
  <arg name="yaw" value="3.1415"/>
  <arg name="world" value="$(find challenge_project)/worlds/challenge2.world"/>
  </include>

   <node name="obstacle_control" pkg="challenge_project" type="challenge2_task2_world_control.py"/>
   <node pkg="challenge_project" name="lds_distance" type="lds_distance_publisher.py" required='true'/>
   <node pkg="challenge_project" name="lds_stop" type="distance_servoing.py" required="true"/>

  <node name="pos_vel_plot"
        pkg="rqt_plot"
        type="rqt_plot"
        args="/lds_distance/N" />

</launch>
